s3fs/core.py

```diff
@@ -215,6 +226,7 @@ def info(self, path):
 
     def walk(self, path):
         """ Return all entries below path """
+        path = path.lstrip('s3://')
```

Comment:
Beware, `lstrip` doesn't work this way.  It strips all occurrences of `s, 3, :, /` from the front.

```python
In [1]: 'Hello'.lstrip('Hel')
Out[1]: 'o'
```

You probably want something like the following:

    path = path[len('s3://'):] if path.startswith('s3://') else path